app/models/outgoing_message.rb

```diff
@@ -455,7 +455,7 @@ def format_of_body
   end
 
   def template_changed
-    if body.empty? || body =~ /\A#{Regexp.escape(letter_template.salutation(default_message_replacements))}\s*#{Regexp.escape(letter_template.letter(default_message_replacements)).gsub(/\//, "\\/").gsub('\n', '')}\s*#{Regexp.escape(letter_template.signoff(default_message_replacements))}/ || body =~ /#{Regexp.escape(Template::InternalReview.details_placeholder)}/
+    if body.empty? || body =~ /\A#{regex_escape(letter_template.body(default_message_replacements))}/ || body =~ /#{Regexp.escape(Template::InternalReview.details_placeholder)}/
```

Comment:
I _think_ this should work without the last check (`|| body =~ /#{Regexp.escape(Template::InternalReview.details_placeholder)}/`) since that would be included in the `letter_template.body` of an internal review.